hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/HazelcastInferTypes.java

```diff
@@ -81,6 +81,18 @@
         }
     };
 
+    public static final SqlOperandTypeInference VARCHAR_IF_UNKNOWN = (callBinding, returnType, operandTypes) -> {
+        RelDataType unknownType = callBinding.getTypeFactory().createUnknownType();
+
+        for (int i = 0; i < operandTypes.length; i++) {
+            RelDataType operandType = operandTypes[i];
+
+            if (operandType == unknownType) {
```

Comment:
.equals?

Reply:
Removed this anonymous class. Now using `ReplaceUnknownOperandTypeInference`, where this specific `==` has been replaced with `equals`.
See https://github.com/hazelcast/hazelcast/pull/17379/commits/6a01e16bd9b3450ef82536e461db30bb9017f1bc